mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -396,31 +396,33 @@ async fn validate_and_store_starting_header(
     ticker: &str,
     storage: &BlockHeaderStorage,
     spv_conf: &SPVConf,
-) -> Result<(), StartingHeaderValidationError> {
+) -> MmResult<(), StartingHeaderValidationError> {
     let height = spv_conf.starting_block_header.height;
     let header_bytes = client
         .blockchain_block_header(height)
         .compat()
         .await
-        .map_err(|e| StartingHeaderValidationError::RpcError(e.to_string()))?;
+        .map_err(|err| MmError::new(StartingHeaderValidationError::RpcError(err.to_string())))?;
```

Comment:
Could you use `map_to_mm`? This allows to avoid using `MmError::new` internally:
```rust
.map_to_mm(|err| StartingHeaderValidationError::RpcError(err.to_string()))?;
```

Same for other places in this file